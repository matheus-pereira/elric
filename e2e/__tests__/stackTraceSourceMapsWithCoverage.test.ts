/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import * as path from 'path';
import wrap from 'elric-snapshot-serializer-raw';
import {extractSummary, runYarnInstall} from '../Utils';
import runelric from '../runelric';

it('processes stack traces and code frames with source maps with coverage', () => {
  const dir = path.resolve(
    __dirname,
    '../stack-trace-source-maps-with-coverage',
  );
  runYarnInstall(dir);
  const {stderr} = runelric(dir, ['--no-cache', '--coverage']);

  // Should report an error at source line 13 in lib.ts at line 10 of the test
  expect(wrap(extractSummary(stderr).rest)).toMatchSnapshot();
});
