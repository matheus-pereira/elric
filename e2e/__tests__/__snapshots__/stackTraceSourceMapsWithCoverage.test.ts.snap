// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`processes stack traces and code frames with source maps with coverage 1`] = `
FAIL __tests__/fails.ts
  ✕ fails

  ● fails

    This did not work!

      13 |
      14 | export function error() {
    > 15 |   throw new Error('This did not work!');
         |         ^
      16 | }
      17 |

      at Object.error (lib.ts:15:9)
      at Object.<anonymous> (__tests__/fails.ts:10:3)
`;
